unstructured/partition/pdf.py

```diff
@@ -557,10 +572,39 @@ def _process_pdfminer_pages(
     laparams = LAParams()
     device = PDFPageAggregator(rsrcmgr, laparams=laparams)
     interpreter = PDFPageInterpreter(rsrcmgr, device)
-
-    for i, page in enumerate(PDFPage.get_pages(fp)):  # type: ignore
-        interpreter.process_page(page)
-        page_layout = device.get_result()
+    try:
+        # Detect invalid dictionary construct for entire PDF
+        pages = list(PDFPage.get_pages(fp))
```

Comment:
Are we calling `list` conversion here to ensure all pages gets processed so we can detect error? but this also means we are holding more data in memory compared to before (when pages is a generator). 
Maybe a cleaner solution is to create a helper like this suggestions here: https://stackoverflow.com/a/27924011

Reply:
> Are we calling list conversion here to ensure all pages gets processed so we can detect error?

Yes, converting to list is needed for error detection, otherwise the error will show up in `for page in PDFPage.get_pages(fp)`